src/Settings/ProjectSettings.ts

```diff
@@ -23,10 +24,30 @@ export class ProjectSettings extends Settings {
     }
 
     /**
-     * Gets a value by key from Azure DevOps.
+     * Gets the value of the given key from the given map of existing Azure DevOps project settings. If the key
+     * does not exist, then the given default value is returned.
      */
-    public get(key: string): PromiseLike<string> {
-        return super.get(this.getProjectSpecificKey(key));
+    public getOrDefault(key: string, existingSettings : Map<string, string>, defaultValue: string): string {
+        const settingKey = this.getProjectSpecificKey(key);
+        if(existingSettings.has(settingKey)){
+            return existingSettings.get(settingKey);
+        }
+        return defaultValue;
+    }
+
+    /** Fetches and returns the stored project settings. */
+    public async loadStoredProjectSettings(): Promise<Map<string, string>> {
+        const storedSettingsMap = new Map<string, string>();
+        const projectPrefix = this.project.substring(0, 30) + '-';
```

Comment:
suggestion (non-blocking)
```suggestion
        const projectPrefix = this.getProjectSpecificKey('');
```